packages/dynamoose/lib/Item.ts

```diff
@@ -30,6 +30,14 @@ export interface ItemSettings {
 	type?: "fromDynamo" | "toDynamo";
 }
 
+export interface ItemMethods {
+	[key: string]: Function;
+}
+
+export interface AnyItemMethods {
+	[key: string]: any;
+}
```

Comment:
It's not immediately clear what this is doing. Seems more general than just `AnyItemMethods`. Why do we have to use `any` vs `Function`? If it's item methods, shouldn't that be a function?

Reply:
Fair point. I encountered an issue with TS early on omitting the `ItemMethods` generic when calling `dynamoose.model`, however I believe that requiring custom methods to extend `ItemMethods` addresses this. Let me revisit this and see if it's possible to completely remove it.